py/builtinimport.c

```diff
@@ -237,6 +242,15 @@ mp_obj_t mp_builtin___import__(mp_uint_t n_args, const mp_obj_t *args) {
     }
     DEBUG_printf("Module not yet loaded\n");
 
+    #if MICROPY_MODULE_FROZEN
+    mp_lexer_t *lex = mp_find_frozen_module(mod_str, mod_len);
+    if (lex != NULL) {
+        module_obj = mp_obj_new_module(MP_OBJ_QSTR_VALUE(module_name));
```

Comment:
Wel, my general assumption that module name is identifier and thus always qstr, but thinking that it can be called as `__import__`, yes, need to do that.